### DISCUSSION TITLE
Training with different resolution images

### DISCUSSION THREAD

**Discussion by Paasu** [neuropoly/axondeepseg#279] [Help needed] [answered]
Am I able to train images that have a different resolution? The images I wish to train and use are 
TEM images at 4008x2672 pixels and 0.0016*um/pixel. Sorry if this is a stupid question, I ask because the training guildline > 1.2 says that I should use images that are 0.01um/pixel. Does that mean I am unable to use my own images. And if I can't, is there any current solution to this?

Thanks.

**Comment by mathieuboudreau** [comment, marked as answer]
Hey @Paasu !

There are no stupid question - don't fret with reaching out if you some parts of our software or user interface isn't clear enough for you, the user.

I just reread that section of [training_guideline.ipynb](https://github.com/neuropoly/axondeepseg/blob/master/notebooks/training_guideline.ipynb), and saw that the "Important" note was not clear in this context. That text is describing the parameters for our pre-saved models, and are not applicable if you are training a new model. We should remove of clarify that text.

**There is no limitation in terms of which resolution you can train your data with.** In fact, the model training algorithm itself never really "knows" your resolution; it just trains on the patched images that it's given, regardless of what their "true resolution" is.

The only two times that resolution is needed in our software is 1- to resample images into patches, and 2- to get morphological statistics (e.g. myelin thickness). For your case, explaining "1" is all that's necessary at the moment.

Before you train your images, you need to split them into workable smaller patches. We typically train on image sizes of 256x256 px or 512x512 px. To do so, you'll first need to use the [guide_dataset_building.ipynb](https://github.com/neuropoly/axondeepseg/blob/master/notebooks/guide_dataset_building.ipynb) notebook. What this notebook does is 1 - it resamples your images into a standard resolution that you can define (i.e. for our own datasets so far, we chose 0.1 and 0.01 um/pixel, because of the next point) since all your images might not have the same resolution, and 2- it splits your images into workable patches (a size that you can define). Now, depending on 1-how large your images are and 2- how big your axons are relative to your image, you might want to choose a "standard resampling resolution" that downsamples your image, so that the model processes faster or most of your axons are not cut through by the patching step.

That's the gist of it. We're close to merging an important branch, so it might be best for you to try running on that branch ([see here](https://github.com/neuropoly/axondeepseg/tree/vs_data_aug_keras)) as it contains some better data augmentation implementations. If my explanations only caused you more confusions, I'd be happy to schedule a video call with you and could walk through some of those steps by using share screen.

Cheers,

Mathieu

**Comment by Paasu** [comment]
Hey @mathieuboudreau

Thank you very much for the prompt reply and I apology for my delayed one.

Your response was clear. My current goal is to train the program with demyelinated axons and axons with decompacted myelin sheaths and compare the results with manual measurement. Perhaps you have other ways I could directly contact you?

Thanks.

**Comment by mathieuboudreau** [comment]
Hi @Paasu  ! 

Have you looked into your issue a bit more and tried some of my tips? If you want to contact me to setup a video call, my email is on my profile: https://github.com/mathieuboudreau

**Comment by Paasu** [comment]
Hi @mathieuboudreau 

Sorry for the late response. I have indeed tried your advice and built my own dataset and have been attempting to train. I have ran into some new issues which I have posted [here](https://github.com/neuropoly/axondeepseg/issues/289). As for the video call, thank you very much for the offer, I will send you an email if I need your assistance. 

Thanks again.

**Comment by SebTim** [comment]
Hi @mathieuboudreau, 

may I ask what are the first/most important parameters that you modulate to optimize new models?
I trained my first model with the original resolution set as "standard resampling resolution" and 256 pixels. Despite GPU assistance, it took 2 days and is not very good, I guess I should have downsampled and used bigger tiles. 
The problem might be as well that the tissue is mouse cortex, so myelin is relatively sparse.

best wishes
